test/y2storage/filesystems/btrfs_test.rb

```diff
@@ -864,4 +864,28 @@
       end
     end
   end
+
+  describe "#copy_mount_by_to_subvolumes" do
+    let(:subvol_mount_points) { filesystem.btrfs_subvolumes.reject { |s| s.mount_point.nil? } }
+    let(:subvol_mount_bys) { subvol_mount_points.map { |m| m.mount_by.to_sym } }
+    let(:btrfs_mount_by) { filesystem.mount_by.to_sym }
+
+    it "starts with both the btrfs and all subvolumes mounted by label" do
```

Comment:
Something like this does not make much sense in RSpec:

```ruby
describe "#whatever_method" do
  it "foo bar" do
    # Something that doesn't call #whatever_method
  end
end
```

Following the convention, if you use a name of a method as title for the "describe", then each "it" block is supposed to describe one aspect of the behavior of such method. So if the method is not called... :thinking: Moreover the output should look like a description of how the method works. Something like this describes how the methods works:

```
Btrfs
  - #copy_mount_by_to_subvolumes
    * copies the mount_by of the Btrfs to all its subvolumes

MountPoint
  - #mount_by=
    * in a Btrfs device
      - sets the mount_by of the Btrfs
      - does not affect the mount_by of the subvolumes
```

Or maybe, to have everything as close as possible:

```
Btrfs
  - #copy_mount_by_to_subvolumes
    * copies the mount_by of the Btrfs to all its subvolumes

  - changing the mount_by with #mount_point.mount_by=
    * sets the mount_by of the Btrfs
    * does not affect the mount_by of the subvolumes
```

But something like this doesn't describe the method Btrfs#copy_mount_by_to_subvolumes (as the two first lines state):
```
Btrfs
  - #copy_mount_by_to_subvolumes
    * starts with both the btrfs and all subvolumes mounted by label
    * changing only the btrfs mount_by does not affect the subvolumes
```

At most, it's a description of the starting point of your test. That's what "let" and "before" are for (setting the starting point). Not the role of "it". And take into account the three "its" could be executed in any order.

Reply:
The point of this is to assert the correct starting conditions. I spent **hours** trying to debug this stuff, and it turned out several times that you cannot assume _anything_ - from no `mount_by` set to having two subvolumes without a mount point to whatever.

Reply:
...all of which just makes _RSpec_ just barf and crash with utterly useless error messages that are completely undebuggable, of course.

Reply:
If you want to ensure the starting condition for all the "its", you can use `expect` in a "before" block. So you can sleep well because you ensure the starting point. That's declarative enough without having to misuse "it".

Reply:
Ok then.